File: 프로그래머스/1/12934. 정수 제곱근 판별/정수 제곱근 판별.cs
using System;

public class Solution {
    public long solution(long n) {
        long answer = 0;
        long num = (long)Math.Sqrt(n);
        if (num * num == n) {
            answer = (num + 1) * (num + 1);
        } else {
            answer = -1;
        }
        return answer;
    }
}

File: 프로그래머스/1/12934. 정수 제곱근 판별/정수 제곱근 판별.java
class Solution {
    public long solution(long n) {
        long answer = 0;
        long sqrt = (long)Math.sqrt(n);
        if ((long)Math.pow(sqrt, 2) == n) {
            answer = (long)Math.pow(sqrt + 1, 2);
        } else {
            answer = -1;
        }
        return answer;
    }
}

/*
 *  sqrt 한 값을 이용하여 Math.pow() 한 값이 n과 동일하면 n+1의 제곱을 리턴, 아니면 -1
*/